src/gui/accountstate.cpp

```diff
@@ -483,6 +487,63 @@ AccountApp* AccountState::findApp(const QString &appId) const
     return nullptr;
 }
 
+void AccountState::connectWebSocket()
+{
+    if (webSocket.state() != QAbstractSocket::UnconnectedState)
+        return;
+
+    const auto &capabilites = _account->capabilities();
+
+    // Check for web socket capability
+    // TODO: Maybe make this more finer
+    if (!capabilites.pushNotificationFilesWebSocketAvailable())
```

Comment:
I wouldn't have pushNotificationFilesWebSocketAvailable and just check if the URL is valid since it's all or nothing for now anyway. Otherwise I'd have a single method return a QFlag instead of having one bool method per type of notification (it would get out of control down the line as more types are added).